src/AEG/CutNode.ts

```diff
@@ -123,36 +123,48 @@ export class CutNode {
-            let isSmallest = true;
-
-            for (let i = 0; i < this.internalChildren.length; i++) {
-                //Check if the point is within a child
-                if (this.internalChildren[i].containsPoint(incomingPoint)) {
-                    isSmallest = false;
-
-                    if (this.internalChildren[i] instanceof CutNode) {
-                        //If the point is within a cut node, check its children
-                        return (this.internalChildren[i] as CutNode).remove(incomingPoint);
-                    } else {
-                        //If the point is within an atom node, remove it
-                        this.internalChildren = this.internalChildren.splice(i, 1);
-                        return true;
+        if (!this.containsPoint(incomingPoint) || this.ellipse === null) {
+            return false;
+        }
+
+        //We do contain the Point at this stage.
+        for (let i = 0; i < this.children.length; i++) {
+            if (this.children[i].containsPoint(incomingPoint)) {
+                if (
+                    this.children[i] instanceof AtomNode || //If the child is childless, the child must be spliced.
+                    (this.children[i] instanceof CutNode &&
+                        (this.children[i] as CutNode).children.length === 0)
+                ) {
+                    this.children.splice(i, 1);
+                    return true;
+                } else {
+                    //We have a CutNode with more than 0 children.
+                    for (let j = 0; j < (this.children[i] as CutNode).children.length; j++) {
+                        if (
+                            (this.children[i] as CutNode).children[j].containsPoint(incomingPoint)
+                        ) {
+                            //If the child has children, and one of its children contains the Point, recursion time.
+                            if ((this.children[i] as CutNode).children[j] instanceof AtomNode) {
+                                (this.children[i] as CutNode).children.splice(j, 1);
+                            } else {
+                                return (
+                                    (this.children[i] as CutNode).children[j] as CutNode
+                                ).remove(incomingPoint);
+                            }
+                        }
                     }
+                    //Here, we have a CutNode with more than 0 children, none of which contained the Point.
+                    //This is now the lowest node containing the Point, and so, we must remove the child.
+                    this.children.splice(i, 1);
+                    return true;
                 }
             }
-            //if the point is not contained within any children nodes, delete this node
-            if (isSmallest) {
-                //What to do to remove this node?
-                return true;
-            }
         }
-
         return false;
     }
 
```

Comment:
What do you do if the point is contained in the cut, but not in any children of the cut/the cut doesn't have any children?

Reply:
I might be misunderstanding as well but wouldn't this solve that? And for when the cut doesn't have any children lines 137-143 handle that
![image](https://github.com/RAIRLab/Peirce-My-Heart/assets/93628226/6066a6c7-ca85-4917-adf7-530e629bb5ac)

Reply:
Aren't lines 136 - 166 the for loop with stuff happening if any child contain the point? I was wondering what happen if we don't enter that loop/exit that loop without any children containing the point. 
For example, in a situation like in the ss below, the point is within the cut but not within any of the children. So if remove is called on the cut, it would pass through the for loop without anything happening because the if on line 137 would return false
![image](https://github.com/RAIRLab/Peirce-My-Heart/assets/89546165/22fa8953-5b02-4176-94ec-f389b9026502)

Reply:
I'm pretty sure with what we talked about this function was meant to be used on the parent of the item you want to delete, that way you can access the to be deleted child easier. But that does also make me realize that 131-133 shouldn't include logic to remove the sheet since there's really no way to remove the item itself here

Reply:
Ohh that does make more sense then, thank you. I think we do need the ellipse === null check to avoid instance of TS screaming at us that ellipse might be null. That happens if you sometimes if you don't explicitly check. I'm not sure about the containsPoint check though, because ideally wouldn't we always call this assuming it does contain the point?